File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/UserInfoDTO.java
package fpt.aptech.server.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserInfoDTO {
    private Integer id;
    private String email;
    private String phone;
    private String avatarUrl;

    // Thông tin phân quyền
    private String roleName;         // Ví dụ: "Quản trị viên", "Người dùng"
    private String roleCode;         // Ví dụ: "ROLE_ADMIN", "ROLE_USER"
    private Set<String> permissions;  // Danh sách các perCode: "READ_REPORT", "WRITE_TRANSACTION"

    // Thông tin cấu hình hệ thống của người dùng
    private String currencyCode;     // Ví dụ: "VND", "USD"
    private Boolean isLocked;        // Trạng thái tài khoản
}

File: APP MOBILE/smart_money/lib/core/models/user_info_dto.dart
/// Thông tin user đầy đủ (dùng cho /api/user/me).
/// Tương ứng: UserInfoDTO.java (server)
class UserInfoDTO {
  final int id;
  final String? email;
  final String? phone;
  final String? avatarUrl;
  final int? roleId;
  final String? roleName;
  final String? roleCode;
  final Set<String> permissions;
  final String? currencyCode;
  final bool? isLocked;

  const UserInfoDTO({
    required this.id,
    this.email,
    this.phone,
    this.avatarUrl,
    this.roleId,
    this.roleName,
    this.roleCode,
    this.permissions = const {},
    this.currencyCode,
    this.isLocked,
  });

  factory UserInfoDTO.fromJson(Map<String, dynamic> json) {
    return UserInfoDTO(
      id: json['id'] as int,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
      avatarUrl: json['avatarUrl'] as String?,
      roleId: json['roleId'] as int?,
      roleName: json['roleName'] as String?,
      roleCode: json['roleCode'] as String?,
      permissions: json['permissions'] != null
          ? Set<String>.from(json['permissions'] as List)
          : {},
      currencyCode: json['currencyCode'] as String?,
      isLocked: json['isLocked'] as bool?,
    );
  }
}

